### ISSUE TITLE
Unspecified Error with MSAccess database

### ISSUE THREAD

**Issue by isaacabraham** [fsprojects/SQLProvider#340]
Trying out with an MS Access DB - connection string looks something like this: -

``` fsharp
type Database = SqlDataProvider<ConnectionString = @"Data Source=path\to\access-db.accdb;Provider=Microsoft.ACE.OLEDB.12.0", DatabaseVendor = DatabaseProviderTypes.MSACCESS>
```

This actually worked once I installed the (correct version of) Access Database Engine.

However, after an unspecified period of time - and only after I start exploring the database in a script - the declaration of the TP breaks with the error `The type provider 'FSharp.Data.Sql.SqlTypeProvider' reported an error: Unspecified error'`. However, FSI continues to work - it's just in the script file that the error shows up.

I've tried both 32 and 64 bit versions of the engine, and both VSCode and VS2015 - same error always occurs.

I'm not in a position to share the database out unfortunately, but am happy to help diagnose this problem if possible.

**Comment by Thorium** [comment]
What version are you using? The latest?

**Comment by Thorium** [comment]
This is probably Access error of the database file being locked while there are multiple connections open.

**Comment by isaacabraham** [comment]
Latest version (1.0.35). I've made sure Access itself isn't open when working with the database - what should I do?

**Comment by Thorium** [comment]
The problem occurs when SQLProvider closes the connection and tries to reconnect. That's why there is a source code some exceptions compared to other providers that Access connections are left open.

**Comment by isaacabraham** [comment]
Don't understand :-( Wouldn't this mean that everyone using Access sees this error?

**Comment by isaacabraham** [comment]
I think you're right though - basically until I actually do a "Send to Interactive" it's fine. But once FSI gets involved as well, it goes red when I start doing a few commands.

A `.laccdb` file is also created after I open up the script that references the DB.

**Comment by pezipink** [comment]
pinging @bennylynch on this one since he wrote the access implementation and is still being used as far as I understand..

**Comment by Thorium** [comment]
Ok, the problem here is:
 - If we wouldn't use transactions and we have an operation like "1. select item, 2 update values, 3. save item", there is a theoretical possibility that someone else is modifying the item after step 1 and before step 3, causing the item maybe to be in a corrupted state.
 - We would like to use DTC-transactions for CRUD-operations. For DTC, the default scope is "attach to existing or create new if none found". We have option to use ADO-database transactions, yes, but they are not spreading over multiple operations like DTCs are. 
 - If connection wants to join a newly created DTC-transaction then the connection has to be made inside a transaction, i.e. connection has to be closed and reopened to attach it to this transaction.
 - If connection is ever closed, Access starts to raise `Unknown Error`s.
 - In other words: we can select do we want DTC-transactions or do we want working intellisense.

**Comment by Thorium** [comment]
This should be working better now, @isaacabraham can you test do the problem still exists?
